### ISSUE TITLE
Improve timestamp logs with human readable format.

### ISSUE THREAD

**Issue by raju2128** [external-secrets/external-secrets#2768]
**Is your feature request related to a problem? Please describe.**
After we deployed the ESO in my k8s-cluster, we generated the logs. we found the timestamp logs of ESO are not in proper format /cluttered. They are not in human-readable format.
Example: {"level":"error",**"ts":1692291266.732894**,"msg":"Reconciler error","controller":"externalsecret","controllerGroup":"external-secrets.io","controllerKind":"ExternalSecret","ExternalSecret":\{"name":"apikey-mongo-secret","namespace":"apollosec"}

**Describe the solution you'd like**
We want timestamp logs of the ESO to be in proper format.
Example: Mon Oct  9 17:37:48 IST 2023

**Describe alternatives you've considered**
No

**Additional context**
No

**Comment by moolen** [comment]
There's a `zap-time-encoding` flag that allows you to specify the: `Zap time encoding (one of 'epoch', 'millis', 'nano', 'iso8601', 'rfc3339' or 'rfc3339nano')`. Does that work for you?

**Comment by mdraijer** [comment]
I agree with @raju2128 about the readability, but more importantly, our log stack has severe problems with the way timestamps are logged.

We write all logs from all ESO components to one datastream in Elastic. That means that they are stored in the same index and that the format of fields must be fixed.

However, where most components log the timestamp to field `ts`, the format is sometimes epoch+micros and sometimes ISO format date+time. Also, one component logs the timestamp in field `time`.

We are not yet running the latest version, but I understand from the fact that this issue is still open, that things haven't changed here.

| Component | Field name | Example value |
|----|----|----|
| external-secrets | ts | 1703520109.1180224 |
| external-secrets-cert-controller | ts | 1703520109.1180224 |
| external-secrets-operator-catalog | time | "2023-12-25T16:00:39Z"<br/>Note: no JSON formatting here |
| external-secrets-operator-controller-manager | ts | "2023-12-25T16:00:39Z" |
| external-secrets-webhook | ts | 1703520109.1180224 |

@moolen _where_ can I set that `zap-time-encoding` flag?

**Comment by mdraijer** [comment]
Maybe @moolen 's remark is referring to https://github.com/external-secrets/external-secrets/pull/1808

It seems that this is not implemented for all components? Also, ideally you will have the same log format in all components, with just one setting. Plus, this does not deal with the non-JSON layout, with timestamp in a different field, of the operator-catalog.

**Comment by mdraijer** [comment]
In the meantime I found that the operator-catalog and operator-controller-manager components are part of Openshift OLM instead of ESO.

Also, setting `extraArgs.zap-time-encoding=iso8601` for all 3 components indeed makes the timestamp in ISO format and thus equal to the timestamp in those OLM components, resolving the problem in Elastic datastreams.

Leaves one or two things to be sorted out:
* Document this feature
* Optional: add a specific parameter in the Helm chart for this feature

**Comment by evrardj-roche** [comment]
This issue has been marked as Stale and will be closed soon if there is no activity. Please update the issue or engage with it to keep it open. It is fine to re-open this issue in the future if you want to work on this.

**Comment by evrardj-roche** [comment]
Hello. Thank you for taking the time to open this issue and contribute to the project.
We truly appreciate community involvement and the effort that goes into reporting issues and suggesting improvements.

We're closing this (Stale) issue as part of a maintenance effort to keep our issue tracker focused and actionable.
Issues that have been inactive for an extended period can create clutter, make it harder to prioritize current work, and distract maintainers from issues that have active contributors ready to work on them.

This doesn't mean your issue isn't valid or important.
If you or anyone else is still affected by this issue and willing to contribute toward a solution (whether through discussion, testing, or code),
please feel free to comment here, reopen the issue, and mention you want to be assigned on this issue.

Alternatively, you can directly submit a PR with `Closes <PR NUMBER>` in its content.

We're happy to reopen or revisit this if there's renewed interest and someone willing to champion it forward.

Thanks again for being part of our community!
